Quarkus Kubernetes extension conflict with Micrometer extension

Quarkus 1.9.1.CR1

**Describe the bug**
When I add Micrometer extension to existing Quarkus project with S3 extension, I am getting below compilation error
`java
Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors [ERROR] [error]: Build step io.quarkus.deployment.index.ApplicationArchiveBuildStep#build threw an exception: java.lang.RuntimeException: Could not resolve artifact org.eclipse.microprofile.metrics:microprofile-metrics-api::jar among the runtime dependencies of the application [ERROR] at 
`
**Expected behavior**
I should be able to compile the project successfully with both S3 extension and MicroMeter extension

**Actual behavior**
Compilation Error


**To Reproduce**

1. Create a Quarkus Projecct with S3 extension
2. Compile build and test (Success)
3. Add Micrometer extension to the project
4. Compile Project (Failure)



**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: mac os & Linux
 - Output of `java -version`:  Java 11
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.9.1.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):

--- a/extensions/kubernetes/vanilla/deployment/pom.xml
+++ b/extensions/kubernetes/vanilla/deployment/pom.xml
@@ -30,10 +30,6 @@
             <artifactId>quarkus-kubernetes-client-deployment-internal</artifactId>
         </dependency>
         <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-smallrye-metrics-spi</artifactId>
-        </dependency>
-        <dependency>
             <groupId>io.dekorate</groupId>
             <artifactId>kubernetes-annotations</artifactId>
             <classifier>noapt</classifier>

--- a/integration-tests/micrometer-prometheus/pom.xml
+++ b/integration-tests/micrometer-prometheus/pom.xml
@@ -45,6 +45,12 @@
             <artifactId>quarkus-resteasy-jackson</artifactId>
         </dependency>
 
+        <!-- Verify extensions co-exist -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-kubernetes</artifactId>
+        </dependency>
+
         <!-- Test Dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>
@@ -121,6 +127,19 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy-jackson-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-kubernetes-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

